scripts/utils/VanityAddress.s.sol

```diff
@@ -1,62 +1,34 @@
-        // Deploy diamond
         string memory json = vm.readFile(JSON_VANITY_PATH);
         uint256 initInt = json.readUint(string.concat("$.", "init"));
-        uint256 i = initInt;
-        address computedAddress;
-        bool found = false;
-        while (!found && i - initInt < 3000000) {
-            computedAddress = _findDeploymentAddress(
-                bytes32(abi.encodePacked(DEPLOYER, abi.encodePacked(uint96(i)))),
-                initCode
-            );
-            if (uint24(uint160(bytes20(computedAddress)) >> 136) == uint24(0x002222)) {
-                found = true;
-                break;
-            }
-            i = i + 1;
-        }
+        uint256 iterations = 3000000;
+
+        (address computed, uint256 found) = minePrefix(initCode, DEPLOYER, 0x002222, initInt, iterations);
+        console.log("Computed: ", computed);
+        console.log("Found: ", found);
 
-        console.log("found ", found);
-        console.log("i ", i);
-        console.logBytes32(bytes32(abi.encodePacked(DEPLOYER, abi.encodePacked(uint96(i)))));
-        console.log("computedAddress ", computedAddress);
+        // write result to json vanity path
+        json = "";
+        vm.serializeUint(json, "init", found);
```

Comment:
Did you test that it was indeed writing well ion the file?

Reply:
yep 👍 it works good